reconcile/test/test_integrations_manager.py

```diff
@@ -193,81 +204,300 @@ def integrations(resources):
-                },
-                {
-                    "name": "integ3",
-                    "resources": resources,
-                },
+                {"name": "integ2", "resources": resources, "extraArgs": None},
+                {"name": "integ3", "resources": resources, "extraArgs": None},
             ],
         },
     ]
     assert namespaces == expected
+
+
+@pytest.fixture
+def shard_manager() -> intop.IntegrationShardManager:
+    return intop.IntegrationShardManager(
+        aws_accounts=[
+            {"name": "acc-1", "disable": None},
+            {"name": "acc-2", "disable": {"integrations": None}},
+            {"name": "acc-3", "disable": {"integrations": []}},
+            {"name": "acc-4", "disable": {"integrations": ["integ1"]}},
+        ],
+        integration_runtime_meta={
+            "integ1": IntegrationMeta(
+                name="integ1", short_help="", args=["--arg", "--account-name"]
+            ),
+            "integ2": IntegrationMeta(name="integ2", short_help="", args=["--arg"]),
+            "integ3": IntegrationMeta(name="integ3", short_help="", args=[]),
+        },
+    )
+
+
+def test_shard_manager_aws_account_filtering(
+    shard_manager: intop.IntegrationShardManager,
+):
+    assert ["acc-1", "acc-2", "acc-3", "acc-4"] == [
+        a["name"]
+        for a in shard_manager._aws_accounts_for_integration("another-integration")
+    ]
+
+
+def test_shard_manager_aws_account_filtering_disabled(
+    shard_manager: intop.IntegrationShardManager,
+):
+    assert ["acc-1", "acc-2", "acc-3"] == [
+        a["name"] for a in shard_manager._aws_accounts_for_integration("integ1")
+    ]
+
+
+def test_shard_manager_supported_args(
+    shard_manager: intop.IntegrationShardManager,
+):
+    assert shard_manager._integration_supports_arg("integ1", "--account-name")
+    assert not shard_manager._integration_supports_arg(
+        "integ1", "--something-unsupported"
+    )
+
+
+@pytest.fixture
+def collected_namespaces_env_test1(
+    integrations: Iterable[Mapping[str, Any]]
+) -> list[dict[str, Any]]:
+    return intop.collect_namespaces(integrations, "test1")
+
+
+def test_initialize_shard_specs_no_shards(
+    collected_namespaces_env_test1: list[dict[str, Any]],
+    shard_manager: intop.IntegrationShardManager,
+):
+    """
+    this test shows how exactly one shard is created when no sharding has been configured
+    """
+    intop.initialize_shard_specs(collected_namespaces_env_test1, shard_manager)
+    expected = [{"shard_id": 0, "shards": 1, "shard_name_suffix": "", "extra_args": ""}]
+    assert (
+        expected
+        == collected_namespaces_env_test1[0]["integration_specs"][0]["shard_specs"]
+    )
+
+
+def test_initialize_shard_specs_two_shards(
+    collected_namespaces_env_test1: list[dict[str, Any]],
+    shard_manager: intop.IntegrationShardManager,
+):
+    """
+    this test shows how the implicit static sharding strategy creates two shards
```

Comment:
I stayed more minutes than I should have thinking about the `implicit` until I understood. Just in case, maybe `default` is a more used term for what I think you mean

```suggestion
    this test shows how the default static sharding strategy creates two shards
```